Example/Swift/ViewController.swift

```diff
@@ -188,31 +170,20 @@ class ViewController: UIViewController, MGLMapViewDelegate, AVSpeechSynthesizerD
         }
     }
     
-    func speak(_ text: String) {
-        let utterance = AVSpeechUtterance(string: text)
-        speechSynth.delegate = self
-        speechSynth.speak(utterance)
-    }
-    
     func startNavigation(_ route: Route) {
-        let camera = mapView.camera
-        camera.pitch = 40
-        mapView.setCamera(camera, animated: false)
-        mapView.userTrackingMode = .followWithCourse
-        navigation = RouteController(route: route)
-        navigation?.resume()
-    }
-    
-    func endNavigation() {
-        instructionView.isHidden = true
-        toggleNavigationButton.isHidden = true
-        howToBeginLabel.isHidden = false
-        mapView.userTrackingMode = .none
-        let camera = mapView.camera
-        camera.pitch = 0
-        mapView.setCamera(camera, animated: true)
-        removeRoutesFromMap()
-        navigation?.suspend()
+        // Pass through a
+        // 1. the route the user will take
+        // 2. A `Directions` class, used for rerouting.
+        let controller = NavigationUI.instantiate(route: route, directions: directions)
```

Comment:
`controller` is actually a view controller that gets pushed into the navigation stack which hangs onto it.